### ISSUE TITLE
Stroom booting successfully when there are failed DB migrations

### ISSUE THREAD

**Issue by at055612** [gchq/stroom#3171]
If a DB migration has failed on a previous boot, stroom is ignoring this on subsequent boots and will boot successfully.

**Comment by at055612** [comment]
Changing it so `flyway.migrate()` is now always run and flyway will decide if anything needs to be migrated.

Logs for a run with no migrations needed now look like
```
-----------------------------------------------------------
  Migrating database module: stroom-security
----------------------------------------------------------- 
INFO  2023-01-09T11:44:51.462Z main o.f.c.i.d.DatabaseFactory      Database: jdbc:mysql://localhost:3307/stroom (MySQL 8.0) 
INFO  2023-01-09T11:44:51.467Z main s.d.u.AbstractFlyWayDbModule   stroom-security - Validating existing and pending Flyway DB migration(s) (BASELINE:1, SUCCESS:3) using history table 'security_schema_history' from path stroom/security/impl/db/migration 
INFO  2023-01-09T11:44:51.476Z main o.f.c.i.c.DbValidate           Successfully validated 4 migrations (execution time 00:00.006s) 
INFO  2023-01-09T11:44:51.480Z main o.f.c.i.command.DbMigrate      Current version of schema `stroom`: 07.00.00.002 
INFO  2023-01-09T11:44:51.480Z main o.f.c.i.command.DbMigrate      Schema `stroom` is up to date. No migration necessary. 
INFO  2023-01-09T11:44:51.481Z main s.d.u.AbstractFlyWayDbModule   stroom-security - No Flyway DB migration(s) applied in path stroom/security/impl/db/migration 
```

and for one with pending migrations
```
INFO  2023-01-09T11:46:48.595Z main s.d.u.AbstractFlyWayDbModule   
-----------------------------------------------------------
  Migrating database module: stroom-security
----------------------------------------------------------- 
INFO  2023-01-09T11:46:48.595Z main o.f.c.i.d.DatabaseFactory      Database: jdbc:mysql://localhost:3307/stroom (MySQL 8.0) 
INFO  2023-01-09T11:46:48.600Z main s.d.u.AbstractFlyWayDbModule   stroom-security - Validating existing and pending Flyway DB migration(s) (PENDING:1, BASELINE:1, SUCCESS:2) using history table 'security_schema_history' from path stroom/security/impl/db/migration 
INFO  2023-01-09T11:46:48.605Z main o.f.c.i.c.DbValidate           Successfully validated 4 migrations (execution time 00:00.002s) 
INFO  2023-01-09T11:46:48.609Z main o.f.c.i.command.DbMigrate      Current version of schema `stroom`: 07.00.00.001 
INFO  2023-01-09T11:46:48.609Z main o.f.c.i.command.DbMigrate      Migrating schema `stroom` to version 07.00.00.002 - test failure 
INFO  2023-01-09T11:46:48.627Z main o.f.c.i.command.DbMigrate      Successfully applied 1 migration to schema `stroom` (execution time 00:00.020s) 
INFO  2023-01-09T11:46:48.628Z main s.d.u.AbstractFlyWayDbModule   stroom-security - Successfully applied 1 Flyway DB migrations using history table 'security_schema_history' 
``` 

and for a run with a bad migration script
```
INFO  2023-01-09T11:50:43.858Z main s.d.u.AbstractFlyWayDbModule   
-----------------------------------------------------------
  Migrating database module: stroom-security
----------------------------------------------------------- 
INFO  2023-01-09T11:50:43.858Z main o.f.c.i.d.DatabaseFactory      Database: jdbc:mysql://localhost:3307/stroom (MySQL 8.0) 
INFO  2023-01-09T11:50:43.864Z main s.d.u.AbstractFlyWayDbModule   stroom-security - Validating existing and pending Flyway DB migration(s) (PENDING:1, BASELINE:1, SUCCESS:2) using history table 'security_schema_history' from path stroom/security/impl/db/migration 
INFO  2023-01-09T11:50:43.871Z main o.f.c.i.c.DbValidate           Successfully validated 4 migrations (execution time 00:00.003s) 
INFO  2023-01-09T11:50:43.875Z main o.f.c.i.command.DbMigrate      Current version of schema `stroom`: 07.00.00.001 
INFO  2023-01-09T11:50:43.876Z main o.f.c.i.command.DbMigrate      Migrating schema `stroom` to version 07.00.00.002 - test failure 
WARN  2023-01-09T11:50:43.877Z main o.f.c.i.s.DefaultSqlScriptExecutor DB: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'XXX

SET SQL_NOTES=@OLD_SQL_NOTES' at line 1 (SQL State: 42000 - Error Code: 1064) 
ERROR 2023-01-09T11:50:43.878Z main o.f.c.i.command.DbMigrate      Migration of schema `stroom` to version 07.00.00.002 - test failure failed! Please restore backups and roll back database and code! 
ERROR 2023-01-09T11:50:49.816Z main s.d.u.AbstractFlyWayDbModule   stroom-security - Error migrating database: 
Migration V07_00_00_002__test_failure.sql failed
------------------------------------------------
SQL State  : 42000
Error Code : 1064
Message    : You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'XXX
```

and for a run with previously failed migrations
```
INFO  2023-01-09T11:49:35.348Z main s.d.u.AbstractFlyWayDbModule   
-----------------------------------------------------------
  Migrating database module: stroom-security
----------------------------------------------------------- 
INFO  2023-01-09T11:49:35.348Z main o.f.c.i.d.DatabaseFactory      Database: jdbc:mysql://localhost:3307/stroom (MySQL 8.0) 
INFO  2023-01-09T11:49:35.354Z main s.d.u.AbstractFlyWayDbModule   stroom-security - Validating existing and pending Flyway DB migration(s) (BASELINE:1, SUCCESS:2, FAILED:1) using history table 'security_schema_history' from path stroom/security/impl/db/migration 
ERROR 2023-01-09T11:49:35.362Z main s.d.u.AbstractFlyWayDbModule   stroom-security - Error migrating database: Validate failed: Detected failed migration to version 07.00.00.002 (test failure) 
```

**Comment by at055612** [comment]
Fixed in >v7.0.11
